### PR TITLE
Remove nullable disable from ColorSpace and Color

### PR DESCRIPTION
**Pull request by stefannikolei**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Remove nullable disable from ColorSpace and Color

### CODE REVIEW

File: src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public partial class ColorSpaceConverter
 {
     private static readonly HunterLabToCieXyzConverter HunterLabToCieXyzConverter = new();
 
-    private LinearRgbToCieXyzConverter linearRgbToCieXyzConverter;
+    private LinearRgbToCieXyzConverter? linearRgbToCieXyzConverter;
```
Comment (antonfirsov) [inline src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs:16]: This is a very strange field, changing it's value for different values of parameters passed to `GetLinearRgbToCieXyzConverter` and it's callers. If there are shared instances, they are not thread-safe. @JimBobSquarePants thoughts? Is it some optimization?

**Review by antonfirsov — COMMENTED** [review]
Not an expert in the area, but we should be careful.

File: src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverterOptions.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public class ColorSpaceConverterOptions
     /// <summary>
     /// Gets or sets the chromatic adaptation method used. When <value>null</value>, no adaptation will be performed.
     /// </summary>
-    public IChromaticAdaptation ChromaticAdaptation { get; set; } = new VonKriesChromaticAdaptation();
+    public IChromaticAdaptation? ChromaticAdaptation { get; set; } = new VonKriesChromaticAdaptation();
```
[the remark this answers is not in the document]
Comment (stefannikolei) [reply src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverterOptions.cs:47]: Perhaps we should also add `public bool PerformChromaticAdaptation { get; set; } = true` 
Then we can make this non nullable.

[the remark this answers is not in the document]
Comment (antonfirsov) [reply src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverterOptions.cs:47]: No, I think nullability is fine here, even the docs state that, which I overlooked.

File: src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs:16]: ColorSpaceConverter is not threadsafe. 

Whole thing needs a rework tbh. I copied the design, and it doesn't allow easy expansion or optimization.

Comment (JimBobSquarePants) [reply src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.CieXyz.cs]: It was an optimization hack. I've replaced it with a concurrent dictionary. Should be safe now.

**Review by JimBobSquarePants — APPROVED** [review]

**Pull request merged**